Auth0Tests/WebViewProviderSpec.swift

```diff
@@ -0,0 +1,295 @@
+// WebViewProviderSpec.swift
+
+#if os(iOS)
+import Quick
+import Nimble
+import WebKit
+@testable import Auth0
+
+private let Timeout: NimbleTimeInterval = .seconds(2)
+
+class WebViewProviderSpec: QuickSpec {
+    
+    override class func spec() {
+        var webViewUserAgent: WebViewUserAgent!
+        var callback: WebAuthProviderCallback!
+        var mockViewController: UIViewController!
+        var mockWebView: WKWebView!
+        
+        let authorizeURL = URL(string: "https://auth0.com/authorize?redirect_uri=https://auth0.com/callback")!
+        let redirectURL = URL(string: "https://auth0.com/callback")!
+        let customSchemeRedirectURL = URL(string: "customscheme://auth0.com/callback")!
+        let code = "abc123"
+        let customSchemeURLWithCode = URL(string: "\(customSchemeRedirectURL.absoluteString)?code=\(code)")!
+        
+        beforeEach {
+            callback = { result in }
+            mockViewController = UIViewController()
+            mockWebView = WKWebView()
+        }
+        
+        describe("WebAuthentication extension") {
+            it("should create a WebView provider") {
+                let provider = WebAuthentication.webViewProvider()
+                expect(provider(authorizeURL, { _ in })).to(beAKindOf(WebViewUserAgent.self))
+            }
+
+            it("should use the fullscreen presentation style by default") {
+                let provider = WebAuthentication.webViewProvider()
+                let userAgent = provider(authorizeURL, { _ in }) as! WebViewUserAgent
+                expect(userAgent.viewController.modalPresentationStyle) == .fullScreen
+            }
+
+            it("should set a custom presentation style") {
+                let style = UIModalPresentationStyle.formSheet
+                let provider = WebAuthentication.webViewProvider(style: style)
+                let userAgent = provider(authorizeURL, { _ in }) as! WebViewUserAgent
+                expect(userAgent.viewController.modalPresentationStyle) == .formSheet
+            }
+        }
+        
+        describe("initialization") {
+            it("should initialize with correct parameters") {
```

Comment:
Should we add a test case that uses `returnTo` instead of `redirect_uri`?

Reply:
yes, added it now